python/ray/data/_internal/execution/operators/map_operator.py

```diff
@@ -275,6 +275,13 @@ def _output_ready_callback(task_index, output: RefBundle):
         def _task_done_callback(task_index, inputs):
             # We should only destroy the input bundle when the whole task is done.
             # Otherwise, if the task crashes in the middle, we can't rerun it.
+            refs = [input[0] for input in inputs.blocks]
+            locations = ray.experimental.get_object_locations(refs)
+            for ref in inputs.blocks:
+                self._metrics.spilled += (
+                    ref[1].size_bytes * locations[ref[0]]["times_spilled"]
```

Comment:
Same here. Wonder if it'd make sense so split `inputs.blocks` into two separate attributes like `inputs.block_refs` and `inputs.metadata`, and then you'd do `for block_ref, metadata in zip(inputs.block_refs, inputs.metadata`. Might be easier to read.